cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesWrapper.java

```diff
@@ -0,0 +1,169 @@
+    configStore.createOrUpdate(NAMESPACE, PREFERENCES, config);
+  }
+
+  private void deletePropertiesHelper(String id) {
+    try {
+      configStore.delete(NAMESPACE, PREFERENCES, id);
+    } catch (ConfigNotFoundException e) {
+      //no-op
+    }
+  }
+
+  public Map<String, String> getProperties() {
+    return getPropertiesHelper(generateId());
+  }
+
+  public Map<String, String> getProperties(String namespace) {
+    return getPropertiesHelper(generateId(namespace));
+  }
+
+  public Map<String, String> getProperties(String namespace, String appId) {
+    return getPropertiesHelper(generateId(namespace, appId));
+  }
+
+  public Map<String, String> getProperties(String namespace, String appId, String programType, String programId) {
+    return getPropertiesHelper(generateId(namespace, appId, programType, programId));
+  }
+
+  public Map<String, String> getResolvedProperties() {
+    return getProperties();
+  }
+
+  public Map<String, String> getResolvedProperties(String namespace) {
+    Map<String, String> propMap = getResolvedProperties();
+    propMap.putAll(getProperties(namespace));
+    return propMap;
+  }
+
+  public Map<String, String> getResolvedProperties(String namespace, String appId) {
+    Map<String, String> propMap = getResolvedProperties(namespace);
+    propMap.putAll(getProperties(namespace, appId));
+    return propMap;
+  }
+
+  public Map<String, String> getResolvedProperties(String namespace, String appId, String programType,
+                                                   String programId) {
+    Map<String, String> propMap = getResolvedProperties(namespace, appId);
+    propMap.putAll(getProperties(namespace, appId, programType, programId));
+    return propMap;
+  }
+
+  public void setProperties(Map<String, String> propMap) {
+    setPropertiesHelper(generateId(), propMap);
+  }
+
+  public void setProperties(String namespace, Map<String, String> propMap) {
+    setPropertiesHelper(generateId(namespace), propMap);
+  }
+
+  public void setProperties(String namespace, String appId, Map<String, String> propMap) {
+    setPropertiesHelper(generateId(namespace, appId), propMap);
+  }
+
+  public void setProperties(String namespace, String appId, String programType, String programId,
+                            Map<String, String> propMap) {
+    setPropertiesHelper(generateId(namespace, appId, programType, programId), propMap);
+  }
+
+  public void deleteProperties() {
+    deletePropertiesHelper(generateId());
+  }
+
+  public void deleteProperties(String namespace) {
+    deletePropertiesHelper(generateId(namespace));
+  }
+
+  public void deleteProperties(String namespace, String appId) {
+    deletePropertiesHelper(generateId(namespace, appId));
+  }
+
+  public void deleteProperties(String namespace, String appId, String programType, String programId) {
+    deletePropertiesHelper(generateId(namespace, appId, programType, programId));
+  }
+
+  private String generateId() {
```

Comment:
is it generate or create/get? "generate" feels like it doesn't create it solely from arguments